packages/vuelidate/src/core.js

```diff
@@ -23,38 +23,30 @@ import { computed, reactive, ref, watch } from 'vue'
 /**
  * Sorts the validators for a state tree branch
  * @param {Object<NormalizedValidator|Function>} validations
- * @return {{ rules: Object<NormalizedValidator>, nestedValidators: Object, config: Object }}
+ * @return {{ rules: Ref<Object<NormalizedValidator>>, nestedValidators: Ref<Object>, config: Ref<Object> }}
  */
 function sortValidations (validations) {
-  const validationKeys = Object.keys(validations)
-
-  let rules = {}
-  let nestedValidators = {}
-  let config = {}
-
-  validationKeys.forEach(key => {
-    const v = validations[key]
-
-    switch (true) {
-      // If it is already normalized, use it
-      case isFunction(v.$validator):
-        rules[key] = v
-        break
-      // If it is just a function, normalize it first
-      // into { $validator: <Fun> }
-      case isFunction(v):
-        rules[key] = { $validator: v }
-        break
-      // Catch $-prefixed properties as config
-      case key.startsWith('$'):
-        config[key] = v
-        break
-      // If it doesn’t match any of the above,
-      // treat as nestedValidators state property
-      default:
-        nestedValidators[key] = v
-    }
-  })
+  let rules = ref({})
+  let nestedValidators = ref({})
+  let config = ref({})
+
+  watch(validations, (changedValidations) => {
+    // rules.value = {}
```

Comment:
Don’t we want to reset those?

Reply:
Yes, but it was causing the entire tree to rebuild, so I was trying out things.